command/vendored_plugins.go

```diff
@@ -102,6 +106,170 @@ var VendoredPostProcessors = map[string]packersdk.PostProcessor{
 	"vsphere":              new(vspherepostprocessor.PostProcessor),
 }
 
+// bundledStatus is used to know if one of the bundled components is loaded from
+// an external plugin, or from the bundled plugins.
+//
+// We keep track of this to produce a warning if a user relies on one
+// such plugin, as they will be removed in a later version of Packer.
+var bundledStatus = map[string]bool{
```

Comment:
Nice way of ensuring we only report for plugins that we are bundling now. I thought it might be an issue for any plugin following the old naming convention `packer-<type>-<component>-name` but that is not the case. Nicely done.